### ISSUE TITLE
[HOLD for payment 2024-05-20] CRITICAL: [P2P Distance] [$500] Enable immediate slide scrolling without tap

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#40211]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.62-6
**Reproducible in staging?:** y
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1712975397589869

## Action Performed:
1. Tap green FAB button
2. Select 'track expense'
3. Tap distance tab
4. Touch the screen and pause ever so briefly after touching the screen. 
5. Touch the screen and immediately move
## Expected Result:
Able to move the map with fingers when touching the screen
## Actual Result:
I need to "tap and then slide" (where my finger needs to hold still for maybe 100ms before moving) rather than just "slide" (ie, where my finger can move immediately after touching the map, like in Google Maps).
## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/1793014e-2d15-49d1-b581-b5fe8ef5594e

https://github.com/Expensify/App/assets/38435837/ebcc67b6-c966-4827-b231-4ed349cd39c5

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013a0e30e9fd5243ce</li>
        <li>Upwork Job ID: 1780282719880835072</li>
        <li>Last Price Increase: 2024-05-02</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Contributor | 0</li>
<li>suneox | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @zanyrenney</details>

**Comment by zanyrenney** [comment]
reviewed, assigining external.

**Comment by mollfpr** [comment]
@suneox @shahinyan11 Are you able to reproduce the issue in the emulator iOS and Android?

**Comment by zanyrenney** [comment]
@mollfpr please can you review the proposals?

**Comment by quinthar** [comment]
What are the next steps?  Are we waiting on @mollfpr to pick a proposal?  What's the ETA on that?

**Comment by mollfpr** [comment]
Sorry for the delay. I just got my phone set up to test the issue.

@shahinyan11 Your solution doesn't look good on the web.

![Screenshot 2024-04-22 at 16 19 09](https://github.com/Expensify/App/assets/25520267/9576cc81-9960-4fa6-87ae-11182be5f895)

**Comment by shahinyan11** [comment]
> @shahinyan11 Your solution doesn't look good on the web.

@mollfpr  Are you about UI or functionality. Are you talking about the user interface or functionality? If you mean the UI, then there is a question of a slight change in styles

**Comment by mollfpr** [comment]
@suneox I'm testing your branch but the issue is still there.

https://github.com/Expensify/App/assets/25520267/d2283ff6-e6b2-4c54-ba37-44e28e0441bf

@shahinyan11 Could you please update your proposal to make it work on all platforms?

**Comment by quinthar** [comment]
@shahinyan11 Can you please give an ETA?  Thanks!

**Comment by mollfpr** [comment]
@suneox Yes, I'm testing it on SG S23 Ultra. You can see the tap in the video in frames 0.11 and 0.15 is moving but the map is not.

**Comment by shahinyan11** [comment]
> @shahinyan11 Can you please give an ETA? Thanks!

I will give updates today

**Comment by shahinyan11** [comment]
I have problems with android build. so it can take more time

**Comment by shahinyan11** [comment]
@mollfpr I have one question. I noticed that my change caused MapView to start capturing navigator gestures. Is it ok ?

**Comment by zanyrenney** [comment]
bump @mollfpr please get back to @shahinyan11 on their question. Thanks!

**Comment by quinthar** [comment]
@shahinyan11 can you please elaborate on what these "navigation gestures" are?  Do you think it's a good thing?

**Comment by shahinyan11** [comment]
@quinthar Sometimes navigation between tabs happens when we swipe from left to right on map

**Comment by neil-marcellini** [comment]
Assigning @ikevin127 as C+ [from Slack here](https://expensify.slack.com/archives/C02NK2DQWUX/p1714151947096399?thread_ts=1714151779.402609&cid=C02NK2DQWUX).

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/40211#issuecomment-2059760924)

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/40211#issuecomment-2059760924).I did complex change to achieve ideal result. I think it works good now

**Comment by ikevin127** [comment]
@suneox I tested both your solutions and neither of them work in fixing the issue, even though there might be some small improvement to the capture of the swipe event on the map - if we swipe quick, the issue is still present:

https://github.com/Expensify/App/assets/56457735/8a8c136e-bc6d-41c6-94a6-ac960b792dd6

**Comment by ikevin127** [comment]
@shahinyan11 Your solution seems to work and given that it's only changing the `index.native.tsx` version of the DraggableFlatList it is a good thing because we're not changing logic on other platforms where the issue does not occur.

https://github.com/Expensify/App/assets/56457735/e6adc129-d158-427a-8fa1-dcc476d9a337

There are a few issues with the solution that I need addressed before we can move on:
1. We need some explanation as to what the solution is doing exactly to fix the issue. I'm asking for this because you simply provided the solution code block, without any explanation as to what the solution is doing.
2. I noticed that the `{...responder.panHandlers}` line from your solution is not required for the solution to work. I removed it because your solution did not provide the source of this `responder` prop. Please also explain this part, whether or not it's actually required and why.
3. I noticed that you used a `setTimeout` in your solution, and according to [Code patterns to avoid](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#code-patterns-to-avoid):
> 1. Make sure any `setTimeout`s are absolutely necessary, not just a workaround for a situation that isn’t fully understood (e.g. a race condition).

If you can clear up these things and update your proposal to reflect these clarifications, I'll do a final review to confirm everything works as expected then I'll recommend for assignment.

**Comment by ikevin127** [comment]
Based on latest #expensify-open-source [Slack 🧵 (conversation)](https://expensify.slack.com/archives/C01GTK53T8Q/p1714150832518049): we're currently waiting on @shahinyan11 to clarify their solution and get it to a working state as currently the app crashes when restarting with the solution applied.

Otherwise we're still open to proposals, preferably one that would work as expected based on the above Slack 🧵 conversation. Also, if this is really urgend I'd suggest increasing the bounty! 💰 

cc @quinthar @neil-marcellini @zanyrenney

**Comment by shahinyan11** [comment]
I will have updates untill tomorrow

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/40211#issuecomment-2059760924). @ikevin127  I updated my solution because it had a bottleneck and failed in one particular case

**Comment by shahinyan11** [comment]
@neil-marcellini @ikevin127 Would you mind if I ask for increased compensation here ?

**Comment by ikevin127** [comment]
♻️ Will review the updated proposals again today!

**Comment by ikevin127** [comment]
> @ikevin127 Could you please try my alternative solution at [this branch](https://github.com/suneox/ExpensifyApp/tree/40211-mapbox-confict-scroll-event)? I just sync and test on the latest main it still works well

@suneox I checked out your test branch and as mentioned in my last comment, the capture seems to be improved but the issue is still present if you swipe really fast. The reason why it might look / seem fixed on your side is because you're testing on an emulator using cursor to swipe and because of this you can't replicate really fast swipes as you would on an actual physical device with touch.

I tested this on a physical OnePlus 7T (Android 14) and here's the video of the behaviour (I activated tap visibility), and I tested your branch fix behaviour vs google maps (end of video) and here are the results:

<details>
<summary>Android: Native (OnePlus 7T - Android 14)</summary>

https://github.com/Expensify/App/assets/56457735/fe4d2cd5-0d28-4eb8-83da-cfb0773efe4b

</details>

**Comment by ikevin127** [comment]
> https://github.com/Expensify/App/issues/40211#issuecomment-2059760924. @ikevin127 I updated my solution because it had a bottleneck and destroyed the navigation in one particular case

@shahinyan11 I tested your updated proposal, looks good, it doesn't crash the app anymore upon restart. There are some details related to the solution which I would change but that can be handled during PR.

✅ Here's how the behaviour looks on a physical device compared to google maps:
<details>
<summary>Android: Native (OnePlus 7T - Android 14)</summary>


https://github.com/Expensify/App/assets/56457735/78ba058a-058d-4e13-8ae7-a4ba23b399d1



</details>

> [!warning]
> There's only **1 issue** left with your current solution:
> 1. It breaks the swipe navigation behaviour the first time you mount the Distance tab page.
>
> What happens exactly is because of your solution, if we try to swipe right to get to the Scan tab, precisely swiping in between the bottom of the map and the Next button -> doesn't work anymore. The user needs to actually tap on the Scan tab, then swipe back to the Distance tab and **only then the swipe navigation would work agai**n from the Distance tab. Here's a video to get some visuals in order to better understand the issue:

**Note:** Pay attention to the tap indicator when I swipe.

<details>
<summary>Android: Native (Nav Swipe Issue)</summary>


https://github.com/Expensify/App/assets/56457735/ce698576-b7fb-4c08-8638-b1787249d2ec



</details>

Once this issue is fixed I think we're good for assignment here!

**Comment by ikevin127** [comment]
No fully working proposal so far, still looking for a proposal that fixes the issue without creating regressions.

**Comment by shahinyan11** [comment]
@ikevin127 @suneox After a lot of research and testing many approaches, I came to the conclusion that it is better to leave everything as is or disable the swipe navigation for the distance tab.

**Comment by ikevin127** [comment]
Bumped in #expensify-open-source [Slack 🧵](https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1714667329605869), hopefully we get more eyes on this!

**Comment by ikevin127** [comment]
> Could you please help me quick check on your device at [the new branch](https://github.com/suneox/ExpensifyApp/tree/40211-mapbox-conflict-with-scrollview)?

I will test and let you know!

> @ikevin127 After my research, the PanResponder on [MapView](https://github.com/suneox/ExpensifyApp/blob/b71a2e0e104a2d3ba285d77f80834cee54f3c693/src/components/MapView/MapView.tsx#L233) sometimes has been [terminated](https://github.com/suneox/ExpensifyApp/blob/b71a2e0e104a2d3ba285d77f80834cee54f3c693/src/components/MapView/MapView.tsx#L60) by DraggableFlatList and keyboardDissmissPanResponder use [onMoveShouldSetPanResponderCapture](https://github.com/suneox/ExpensifyApp/blob/b71a2e0e104a2d3ba285d77f80834cee54f3c693/src/components/ScreenWrapper.tsx#L161) at ScreenWraper so we have to use [onStartShouldSetPanResponderCapture](https://github.com/suneox/ExpensifyApp/blob/b71a2e0e104a2d3ba285d77f80834cee54f3c693/src/components/MapView/MapView.tsx#L67) and [onMoveShouldSetPanResponderCapture](https://github.com/suneox/ExpensifyApp/blob/b71a2e0e104a2d3ba285d77f80834cee54f3c693/src/components/MapView/MapView.tsx#L63) for PanResponder on MapView and separate the custom MapView for android to avoid regression.

It makes sense, I was looking forward to a proposal that would separate logic into `index.android.tsx`.
Will test and get back with an answer in 1-2 hours!

**Comment by ikevin127** [comment]
@suneox Your new test branch tests well 🎉 It fixes our issue as per the expected result (see video below).

<details>
<summary>Android: Native (OnePlus 7T - Android 14)</summary>

https://github.com/Expensify/App/assets/56457735/e00d995b-baff-4fc0-9017-4c1f11167386

</details>

Please update your proposal detailing the **WHY** of the proposed changes so we can clearly understand the root cause and how the solution works in fixing our issue.

**Comment by quinthar** [comment]
Woohoo sounds like we've found a solution?  What's the next step and ETA?

**Comment by ikevin127** [comment]
I was asked by @suneox to test the new branch solution from https://github.com/Expensify/App/issues/40211#issuecomment-2091087791 and if working, they mentioned:

> I will remove the debugging log and explain in detail after you test it works well

To which I confirmed that the test branch tests well and the new solution seems to work as expected in https://github.com/Expensify/App/issues/40211#issuecomment-2091330144.

@quinthar The next step here would be for @suneox to update their proposal according to the new solution so that we can all understand the **WHY** of the RCA and how the solution works in fixing the issue.

> [!important]
> If you think we can skip the detailed proposal update and move forward with the current (not very detailed) updated proposal, then I can move on and assign @suneox.

Please let me know, as I'll be out for the day in about 1 hour!

**Comment by ikevin127** [comment]
@suneox's [proposal](https://github.com/Expensify/App/issues/40211#issuecomment-2059726565) LGTM! The root cause was identified and the proposed solution targets and fixes the issue accordingly.

🎀👀🎀 C+ reviewed

**Comment by shahinyan11** [comment]
@suneox @ikevin127 There is no need to add a separate component for Android, just add  `dragHitSlop` prop to the `DraggableList` component

**Comment by ikevin127** [comment]
> @suneox @ikevin127 There is no need to add a separate component for Android, just add `dragHitSlop` prop to the DraggableList component

I think it's safe to split `index.native.tsx` DraggableList into `index.ios.tsx` and `index.android.ts` since on Android we're restructuring the component render by adding a `View`, which on iOS is not needed because the issue is not present.

This is the recommended way for platform specific changes to be safe and avoid breaking the same, working logic on other platforms.

See https://github.com/Expensify/App/issues/34881#issuecomment-1936178091 issue for reference.

**Comment by neil-marcellini** [comment]
> @suneox's [proposal](https://github.com/Expensify/App/issues/40211#issuecomment-2059726565) LGTM! The root cause was identified and the proposed solution targets and fixes the issue accordingly.
> 
> 🎀👀🎀 C+ reviewed

I found the proposal to be a bit unclear and hard to follow, but I trust the thorough testing that was already done, so let's create a PR.

**Comment by shahinyan11** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/40211#issuecomment-2059760924). I updated proposal as I am sure that the changes in approved [proposal](https://github.com/Expensify/App/issues/40211#issuecomment-2059726565) does not make sense and not necessary to have that behavior

**Comment by ikevin127** [comment]
* [x]  [@ikevin127] The PR that introduced the bug has been identified. Link to the PR:

No PR causing this was identified, this was not caused by a recent regression.
* [x]  [@ikevin127] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
* [x]  [@ikevin127] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

No need, as this was an edge case issue which happened only on Android: Native due to differences in touch handler.
* [x]  [@ikevin127] Determine if we should create a regression test for this bug.

No need for regression test here as as this was an edge case issue which happened only on Android: Native due to differences in touch handler.
* [x]  [@ikevin127] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by zanyrenney** [comment]
**payment summary**

paid $500 via upwork to @suneox 
paid $500 via upwork to @ikevin127
